### PR TITLE
Plumb through support for controlling subgroup size in CodeGen

### PR DESCRIPTION
**Pull request by antiagainst**
This commit plumbs through support for controlling subgroup size (i.e., VK_EXT_subgroup_size_control for Vulkan) in CodeGen:

- Added a new subgroup size attribute to HAL ExecutableExportOp, similar to workgroup size there.
- Extended various helper functions to optionally set subgroup size.
- Defined VK_EXT_subgroup_size_control and updated various target triples regarding min/max subgroup size.

This commit just thread support through CodeGen flow; it's not wired up into runtime to use yet, which will happen later.

### CODE REVIEW

**Comment by antiagainst** [comment]
Depends on https://reviews.llvm.org/D138962 and https://reviews.llvm.org/D139071

File: compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp [hunk 1]
```diff
@@ -263,16 +263,28 @@ SmallVector<int64_t> getWorkgroupSize(IREE::HAL::ExecutableExportOp exportOp) {
...
+    return attr.getValue().getSExtValue();
+  }
+  return {};
+}
+
 void setTranslationInfo(IREE::HAL::ExecutableExportOp exportOp,
                         IREE::Codegen::TranslationInfoAttr translationInfo,
-                        ArrayRef<int64_t> workgroupSize) {
+                        ArrayRef<int64_t> workgroupSize,
+                        Optional<int64_t> subgroupSize) {
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp:277]: I am wondering if you even want to make it part of this. I did this, but I dont think it was a good idea to start with that `workgroupSize`  was part of this. I used it as a crutch, that might not be worth it. COuld we instead set the subgroup size at call site. Bonus points to moving setting `workgroupSize` also out of this method

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp (hunk 1 above)
Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp:279]: Would you mind to elaborate a bit more on `set the subgroup size at call site`? Do you mean having `setTranslationInfo` only setting `IREE::Codegen::TranslationInfoAttr`, and set workgroup/subgroup size separately on the export op?

Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp:279]: I guess that's what you meant. SG; let me move it to separate utility functions.

Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp:279]: https://github.com/iree-org/iree/pull/11400 to split translation info and workgroup size setting.

Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/LoweringConfig.cpp]: Also split the subgroup setting out.

**Review by MaheshRavishankar — APPROVED** [review]
Maybe we should even consider splitting `setDispatchConfig` (or dropping some of these helper functions all together). But unblocking for now.

**Comment by antiagainst** [comment]
> Maybe we should even consider splitting `setDispatchConfig` (or dropping some of these helper functions all together). But unblocking for now.

SG. Let me take another round to clean some stuff up.

**Pull request merged**